vllm/transformers_utils/configs/mpt.py

```diff
@@ -0,0 +1,74 @@
+# Adapted from
+# https://huggingface.co/mosaicml/mpt-7b/blob/main/configuration_mpt.py
```

Comment:
Not sure I correctly understand your question, but MPT is not supported by hf transformers yet. The files are in hf model hub.

Reply:
It's not easy, because those config classes might not follow huggingface's naming conventions.

For example, `MPTConfig` uses `n_head` for the number of attention heads in the model. This is problematic since our `ModelConfig` uses `hf_config.num_attention_heads`, which works for other HF official models. So I added the name mapping `{"num_attention_heads": "n_head"} in our version of `MPTConfig`.

Another example is Falcon. In Falcon, the 7B and 40B models have slightly different config classes. To unify the config classes, we rewrote them to our version (#321).

In general, arbitrary unofficial HF model/config code will not be compatible with vLLM. I feel it's inevitable to rewrite them in vLLM.